awscrt/s3.py

```diff
@@ -222,13 +229,17 @@ def __init__(
             multipart_upload_threshold=None,
             throughput_target_gbps=None,
             enable_s3express=False,
-            memory_limit=None):
+            memory_limit=None,
+            network_interface_names=None):
         assert isinstance(bootstrap, ClientBootstrap) or bootstrap is None
         assert isinstance(region, str)
         assert isinstance(signing_config, AwsSigningConfig) or signing_config is None
         assert isinstance(credential_provider, AwsCredentialsProvider) or credential_provider is None
         assert isinstance(tls_connection_options, TlsConnectionOptions) or tls_connection_options is None
         assert isinstance(part_size, int) or part_size is None
+        assert isinstance(network_interface_names, list) and all(isinstance(name, str)
+                                                                 for name in network_interface_names) or network_interface_names is None
```

Comment:
you don't NEED to add these checks, since we check the type down in C, where we're going to accept lists OR tuples

we're not super consistent with these type-checks here in python code